examples/java/src/main/java/org/apache/beam/examples/complete/TopWikipediaSessions.java

```diff
@@ -76,7 +77,12 @@
     @ProcessElement
     public void processElement(ProcessContext c) {
       TableRow row = c.element();
-      int timestamp = (Integer) row.get("timestamp");
+      int timestamp;
+      try {
+        timestamp = ((BigDecimal) row.get("timestamp")).intValue();
+      } catch (ClassCastException e) {
+        timestamp = ((Integer) row.get("timestamp")).intValue();
+      }
```

Comment:
Dataflow converts into `BigDecimal`, while Direct runner converts into `Integer`.

Reply:
Can you elaborate on this issue a bit please? Do we have similar issue or different types for other runners?

Reply:
@apilloud do you know why this happens on parsing of JSON files?

Reply:
If you're thinking the `TableRow` here is related to SQL, it isn't. This is a BigQuery datatype. The JSON parsing is being configured to use a global instance of Jackson in `ParseTableRowJson`, later in this file. The `TableRow` type is really just `Map<String,Object>`, so Jackson will pick a type for numbers based on global config which must not be the same on all runners. (See https://fasterxml.github.io/jackson-databind/javadoc/2.8/index.html?com/fasterxml/jackson/databind/DeserializationFeature.html for some of these config options.) One way to fix this would be to provide the actual types for json decoding rather then using `Map<String,Object>`.